crates/polars-core/src/chunked_array/ops/apply.rs

```diff
@@ -14,35 +14,95 @@ impl<T> ChunkedArray<T>
 where
     T: PolarsDataType,
 {
-    pub fn apply_values_generic<'a, U, K, F>(&'a self, op: F) -> ChunkedArray<U>
+    pub fn apply_values_generic<'a, U, K, F>(&'a self, mut op: F) -> ChunkedArray<U>
     where
         U: PolarsDataType,
-        F: FnMut(T::Physical<'a>) -> K + Copy,
+        F: FnMut(T::Physical<'a>) -> K,
         U::Array: ArrayFromIter<K>,
     {
         let iter = self.downcast_iter().map(|arr| {
-            let out: U::Array = arr.values_iter().map(op).collect_arr();
+            let out: U::Array = arr.values_iter().map(&mut op).collect_arr();
             out.with_validity_typed(arr.validity().cloned())
         });
 
         ChunkedArray::from_chunk_iter(self.name(), iter)
     }
 
-    pub fn try_apply_values_generic<'a, U, K, F, E>(&'a self, op: F) -> Result<ChunkedArray<U>, E>
+    pub fn try_apply_values_generic<'a, U, K, F, E>(
+        &'a self,
+        mut op: F,
+    ) -> Result<ChunkedArray<U>, E>
     where
         U: PolarsDataType,
-        F: FnMut(T::Physical<'a>) -> Result<K, E> + Copy,
+        F: FnMut(T::Physical<'a>) -> Result<K, E>,
         U::Array: ArrayFromIter<K>,
     {
         let iter = self.downcast_iter().map(|arr| {
-            let element_iter = arr.values_iter().map(op);
+            let element_iter = arr.values_iter().map(&mut op);
             let array: U::Array = element_iter.try_collect_arr()?;
             Ok(array.with_validity_typed(arr.validity().cloned()))
         });
 
         ChunkedArray::try_from_chunk_iter(self.name(), iter)
     }
 
+    // Applies a function only to the non-null elements, propagating nulls.
```

Comment:
I think we should disambiguate this with `apply_values_generic`. E.g. one ignores nulls and applies to all values (including the masked out ones) and reapplies the nulls at the end. The other creates a branch.

Reply:
I'm up for naming suggestions, we should have both though.

Reply:
Documenting the difference seems good to me.